Products/ZenUtils/tests/testExceptions.py

```diff
@@ -13,30 +21,63 @@ def __init__(self):
     def raiseException(self):
         raise self.raise_value, self.type, self.tb
 
-class ExceptionTestCases(BaseTestCase):
-    def test_resolveException_with_exception_value(self):
-        failure = Failure()
-        failure.value = ZentinelException()
-        self.assertIs( failure.value, resolveException( failure))
+
+class ExceptionTestCases(TestCase): #BaseTestCase):
+
+    def test_resolveException(self):
+        exception = RuntimeError('genric exception')
+        twisted_failure = Failure(exception)
+        out = resolveException(twisted_failure)
+        self.assertIs(out, exception)
 
     def test_resolveException_with_exception_string(self):
-        failure = Failure()
+        '''LEGASY: twisted.python.Failure requires a valid exception
+        '''
+        failure = DummyFailure()
         failure.type = "Products.ZenUtils.Exceptions.ZenPathError"
-        self.assertIsInstance( resolveException( failure), ZenPathError)
+        out = resolveException(failure)
+        expected = ZenPathError(failure.value, failure.tb)
+        self.assertIsInstance(out, ZenPathError)
+        self.assertEqual(out.args, expected.args)
+        self.assertEqual(out.message, expected.message)
 
     def test_resolveException_with_exception_string_with_import_error(self):
-        failure = Failure()
+        '''LEGASY: twisted.python.Failure requires a valid exception                                 
+        '''
+        failure = DummyFailure()
         failure.value = "value"
         failure.tb = "tb"
         failure.type = "Products.ZenUtils.Exceptions.UnknownError"
-        actual = resolveException( failure)
-        self.assertIsInstance( actual, Exception)
-        self.assertEquals( ("value", "tb"), actual.args)
+        out = resolveException(failure)
+        expected = ZenResolveExceptionError(
+            failure.value, failure.tb, failure.__dict__
+        )
+        self.assertIsInstance(out, Exception)
+        self.assertEquals(out.args, ("value", "tb", failure.__dict__))
 
     def test_resolveException_with_raiseException(self):
-        failure = Failure()
+        failure = DummyFailure()
         failure.raise_value = RuntimeError()
-        self.assertIs( failure.raise_value, resolveException( failure))
+        self.assertIs(failure.raise_value, resolveException(failure))
+
+    def test_resolveException_handles_twisted_spread_pb_RemoteError(self):
+        error = RemoteError('remoteType', 'value', 'remoteTraceback')
+        failure = Failure(error)
+        #failure.type = 'twisted.spread.pb.RemoteError'
```

Comment:
If this is supposed to be informative, it needs more words, or it needs to be removed.